mkosi/qemu.py

```diff
@@ -938,22 +941,44 @@ def run_qemu(args: Args, config: Config) -> None:
 
                 kcl += [root]
             elif config.output_format == OutputFormat.directory:
-                sock = stack.enter_context(start_virtiofsd(config, fname, uidmap=False))
+                sock = stack.enter_context(
+                    start_virtiofsd(
+                        config,
+                        fname,
+                        name=config.machine_or_name(),
+                        selinux=bool(want_selinux_relabel(config, fname, fatal=False))),
+                )
                 cmdline += [
                     "-chardev", f"socket,id={sock.name},path={sock}",
                     "-device", f"vhost-user-fs-pci,queue-size=1024,chardev={sock.name},tag=root",
                 ]
                 kcl += ["root=root", "rootfstype=virtiofs", "rw"]
 
-        for tree in config.runtime_trees:
-            sock = stack.enter_context(start_virtiofsd(config, tree.source, uidmap=True))
-            tag = tree.target.name if tree.target else tree.source.name
+        def add_virtiofs_mount(sock: Path, dst: PathString, *, tag: str) -> None:
+            nonlocal cmdline, kcl
```

Comment:
It's a big uglier to pass in `cmdline` and kcl` as arguments and return them, but I think it's worth it, since it's easier to reason about than `nonlocal`.